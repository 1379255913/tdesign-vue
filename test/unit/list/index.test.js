import { mount } from '@vue/test-utils';
import { List } from '@/src/list/index.ts';

describe('List', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(List, {
      propsData: {
        header: 'header',
        footer: 'footer',
        asyncLoading: 'loading',
        size: 'large',
      },
    });
  });

  it('equals header to "header"', () => {
    expect(cmp.vm.header).toEqual('header');
  });
  it('equals footer to "footer"', () => {
    expect(cmp.vm.footer).toEqual('footer');
  });
  it('equals asyncLoading to "loading"', () => {
    expect(cmp.vm.asyncLoading).toEqual('loading');
  });
  it('equals size to "large"', () => {
    expect(cmp.vm.size).toEqual('large');
  });
  // 待设计稿输出
  // it('equals layout to "horizontal"', () => {
  //   expect(cmp.vm.layout).toEqual('horizontal');
  // });
});
