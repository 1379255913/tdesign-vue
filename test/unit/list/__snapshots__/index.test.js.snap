// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List :props :asyncLoading is a function, () => VNode. 1`] = `
<div class="t-list t-size-m">
  <ul class="t-list-items"></ul>
  <div class="t-list__load">
    <p>This is a loading</p>
  </div>
</div>
`;

exports[`List :props :asyncLoading is a string 1`] = `
<div class="t-list t-size-m">
  <ul class="t-list-items">text</ul>
  <div class="t-list__load t-list__load--loading">
    <div><svg viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg" class="t-icon t-icon-loading">
        <path fill-opacity=".9" fill-rule="evenodd" d="M96 512C96 282.264 282.264 96 512 96 741.736 96 928 282.264 928 512L824 512 824 512C824 339.68715776 684.3128448 200 512 200 339.68715776 200 200 339.68715776 200 512 200 684.3128448 339.68715776 824 512 824L512 928C282.264 928 96 741.736 96 512Z"></path>
      </svg><span>正在加载中，请稍等</span></div>
  </div>
</div>
`;

exports[`List :props :asyncLoading is loading-more 1`] = `
<div class="t-list t-size-m">
  <ul class="t-list-items">text</ul>
  <div class="t-list__load t-list__load--loading-more"><span>点击加载更多</span></div>
</div>
`;

exports[`List :props :footer is a function, () => VNode. 1`] = `
<div class="t-list t-size-m">
  <ul class="t-list-items"></ul>
  <div class="t-list__footer">
    <p>This is a footer</p>
  </div>
  <div class="t-list__load"></div>
</div>
`;

exports[`List :props :footer is a string 1`] = `
<div class="t-list t-size-m">
  <ul class="t-list-items">text</ul>
  <div class="t-list__footer">footer</div>
  <div class="t-list__load"></div>
</div>
`;

exports[`List :props :header is a function, () => VNode. 1`] = `
<div class="t-list t-size-m">
  <div class="t-list__header">
    <p>This is a header</p>
  </div>
  <ul class="t-list-items"></ul>
  <div class="t-list__load"></div>
</div>
`;

exports[`List :props :header is a string 1`] = `
<div class="t-list t-size-m">
  <div class="t-list__header">header</div>
  <ul class="t-list-items">text</ul>
  <div class="t-list__load"></div>
</div>
`;

exports[`List :props :layout 1`] = `
<div class="t-list t-size-m">
  <ul class="t-list-items">text</ul>
  <div class="t-list__load"></div>
</div>
`;

exports[`List :props :size 1`] = `
<div class="t-list t-size-l">
  <ul class="t-list-items">text</ul>
  <div class="t-list__load"></div>
</div>
`;

exports[`List :props :split 1`] = `
<div class="t-list t-size-m t-list--split">
  <ul class="t-list-items">text</ul>
  <div class="t-list__load"></div>
</div>
`;

exports[`List :props :stripe 1`] = `
<div class="t-list t-size-m">
  <ul class="t-list-items">text</ul>
  <div class="t-list__load"></div>
</div>
`;

exports[`ListItem :props :action 1`] = `
<li class="t-list-item">
  <div class="t-list-item-main">text
<li class="t-list-item__action">action</li>
</div>
</li>
`;

exports[`ListItem :props :action is a function, () => VNode. 1`] = `
<li class="t-list-item">
  <div class="t-list-item-main">
<li class="t-list-item__action">
  <p>This is a action</p>
</li>
</div>
</li>
`;

exports[`ListItem :props :content 1`] = `
<li class="t-list-item">
  <div class="t-list-item-main">text</div>
</li>
`;

exports[`ListItem :props :content is a function, () => VNode. 1`] = `
<li class="t-list-item">
  <div class="t-list-item-main">
    <p>This is a content</p>
  </div>
</li>
`;

exports[`ListItem :props :default 1`] = `
<li class="t-list-item">
  <div class="t-list-item-main">default</div>
</li>
`;

exports[`ListItem :props :default is a function, () => VNode. 1`] = `
<li class="t-list-item">
  <div class="t-list-item-main">
    <p>default</p>
  </div>
</li>
`;

exports[`ListItemMeta :props :avatar 1`] = `
<div class="t-list-item__meta">
  <div class="t-list-item__meta-avatar"><img src="avatar" alt=""></div>
  <div class="t-list-item__meta-content"></div>
</div>
`;

exports[`ListItemMeta :props :avatar is a function, () => VNode. 1`] = `
<div class="t-list-item__meta">
  <div class="t-list-item__meta-avatar"></div>
  <div class="t-list-item__meta-content"></div>
</div>
`;

exports[`ListItemMeta :props :description 1`] = `
<div class="t-list-item__meta">
  <div class="t-list-item__meta-avatar"></div>
  <div class="t-list-item__meta-content">
    <p class="t-list-item__meta-description">description</p>
  </div>
</div>
`;

exports[`ListItemMeta :props :description is a function, () => VNode. 1`] = `
<div class="t-list-item__meta">
  <div class="t-list-item__meta-avatar"></div>
  <div class="t-list-item__meta-content">
    <p class="t-list-item__meta-description">
    <p>description</p>
    </p>
  </div>
</div>
`;

exports[`ListItemMeta :props :title 1`] = `
<div class="t-list-item__meta">
  <div class="t-list-item__meta-avatar"></div>
  <div class="t-list-item__meta-content">
    <h3 class="t-list-item__meta-title">title</h3>
  </div>
</div>
`;

exports[`ListItemMeta :props :title is a function, () => VNode. 1`] = `
<div class="t-list-item__meta">
  <div class="t-list-item__meta-avatar"></div>
  <div class="t-list-item__meta-content">
    <h3 class="t-list-item__meta-title">
      <p>title</p>
    </h3>
  </div>
</div>
`;
